pandas/core/internals/managers.py

```diff
@@ -808,6 +810,9 @@ def as_array(self, transpose: bool = False) -> np.ndarray:
         else:
             arr = self._interleave()
 
+        if na_value is not lib.no_default:
+            arr[isna(arr)] = na_value
```

Comment:
Suggestion: pass through this `na_value` to `_interleave`, and then when looping over the blocks, you can check for ExtensionBlocks and then call `to_numpy` with the appropriate dtype and na_value. 

The lines above will need to be kept for non-EAs, but I think for EAs, we want to use their `to_numpy` implemenation, as this can potentially be more efficient in depending on the dtype/na_value combination

Reply:
Pushed a change to make this update. In the single extension block case we're still doing a conversion to object (got several odd failures when I tried making the associated edit there) but I'm thinking this is already going to be a pretty inexpensive case anyways

Reply:
> In the single extension block case we're still doing a conversion to object 

I think in the above `is self._is_single_block` checks, you can add an additional case for single ExtensionBlock (like there is already a check for single datetimetz block)

Reply:
Got mixed up by the "blocks are transposed" thing and was missing a reshape, should be working now 🤞